platform/kube/inject/inject.go

```diff
@@ -236,41 +222,47 @@ func GetInitializerConfig(kube kubernetes.Interface, namespace, injectConfigName
 }
 
 func injectRequired(namespacePolicy InjectionPolicy, obj metav1.Object) bool {
-	var resourcePolicy string
+	var useDefault bool
+	var inject bool
 
 	annotations := obj.GetAnnotations()
 	if annotations == nil {
-		resourcePolicy = istioSidecarAnnotationPolicyValueDefault
+		useDefault = true
 	} else {
-		var ok bool
-		if resourcePolicy, ok = annotations[istioSidecarAnnotationPolicyKey]; !ok {
-			resourcePolicy = istioSidecarAnnotationPolicyValueDefault
+		if value, ok := annotations[istioSidecarAnnotationPolicyKey]; !ok {
+			useDefault = true
+		} else {
+			// http://yaml.org/type/bool.html
+			switch strings.ToLower(value) {
+			case "y", "yes", "true", "on":
```

Comment:
may be add enabled ?

Reply:
I would prefer to limit the number of valid values if possible. http://yaml.org/type/bool.html seemed like a reasonable compromise and is already standardized. If we can always extend later if necessary without backwards compat issues.